src/pybind/mgr/dashboard/frontend/src/app/shared/services/text-to-download.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+
+import { saveAs } from 'file-saver';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TextToDownloadService {
+  constructor() {}
+
+  download(downloadText: string, filename?: string) {
```

Comment:
As this is pretty static you can just export the function instead.

```suggestion
  export function saveTextAsFile(text: string, filename?: string) {
```

Reply:
Hmmm I'm not sure... I don't see a huge benefit in changing it... and I assume it will be harder to test.